src/CalcViewModel/StandardCalculatorViewModel.cpp

```diff
@@ -209,6 +212,25 @@ void StandardCalculatorViewModel::SetPrimaryDisplay(_In_ wstring const& displayS
     {
         UpdateProgrammerPanelDisplay();
     }
+
+    AreAlwaysOnTopResultsUpdated = false;
+    if (!IsEditingEnabled && m_tokens != nullptr)
+    {
+        unsigned int nTokens = 0;
+        m_tokens->GetSize(&nTokens);
+        DoesAlwaysOnTopResultShowTokens = !IsInError;
+        if (IsEngineRecording || nTokens == 0 || IsInError)
+        {
+            DoesAlwaysOnTopResultConcatenate = true;
+            CalculationAlwaysOnTopResultAutomationName = GetCalculatorExpressionAutomationName() + " " + CalculationResultAutomationName;
```

Comment:
Similar to my other comment - concatenating strings usually breaks right-to-left languages. To achieve this result, we should create a localizable formatting string in the .resw file.